examples/jsm/objects/MarchingCubes.js

```diff
@@ -890,84 +835,33 @@ class MarchingCubes extends ImmediateRenderObject {
 						const fx = ( x - this.halfsize ) / this.halfsize; //+ 1
 						const q = y_offset + x;
 
-						polygonize( fx, fy, fz, q, this.isolation, renderCallback );
+						 polygonize( fx, fy, fz, q, this.isolation );
 
 					}
 
 				}
 
 			}
 
-			this.end( renderCallback );
+			// reset unneeded data
 
-		};
+			for ( let i = this.count * 3; i < this.positionArray.length; i ++ ) {
 
-		this.generateGeometry = function () {
+				this.positionArray[ i ] = 0.0;
```

Comment:
Instead of resetting the vertices like before, I was trying to use `setDrawRange()` to limit the rendered data, However, this introduces strange flickering (it seems the number of rendered triangles varies from frame to frame). Hence, I've sticked to the previous approach for now.